pkg/util/metrics/system/common.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package system
+
+import (
+	"bufio"
+	"os"
+)
+
+// ReadFile reads contents from a file and returns lines.
+func ReadFile(filename string) ([]string, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	var result []string
+	s := bufio.NewScanner(file)
+	for s.Scan() {
```

Comment:
Why not result the scanner and let the caller Scan a line at a time rather than load everything in memory at once?

Reply:
as /proc/* are smaller files, i thought loading everything at once would be fine and faster and reading from /proc/* files is faster. And these are frequent calls keeping the file open till all the lines are processed i am not sure how does npd would react.

Reply:
Since it's a public func, I'm thinking of a case where we'd have dozens of stats reading various proc files. The memory usage can add up (again, not crazy but the performance gains are sufficiently easy to achieve here).

What are your concerns with open files? I can imagine some racy behavior with an open file while the kernel is writing to it (so we may have inconsistent data).

Not a blocker for this PR though.